projects/framework-premiere/extensions/plugins/premiere_project_saved_validator.py

```diff
@@ -0,0 +1,44 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+
+from ftrack_framework_core.plugin import BasePlugin
+from ftrack_framework_core.exceptions.plugin import (
+    PluginExecutionError,
+    PluginValidationError,
+)
+
+from ftrack_utils.rpc import JavascriptRPC
+
+
+class PremiereProjectSavedValidatorPlugin(BasePlugin):
+    '''
+    Plugin to validate if the Premiere project has been saved.
+    '''
+
+    name = 'premiere_project_saved_validator'
+
+    def run(self, store):
+        '''
+        Run the validation process for the Premiere project.
+        '''
+        component_name = self.options.get('component', 'main')
+
+        # Premiere does not support checking if document is saved, so we will
+        # save the document instead.
+        # Note: Premiere cannot have unsaved projects - they always have a path.
```

Comment:
But then the premiere premiere_save_to_temp_finalizer plugin is not needed because it is always saved somehow?

Reply:
Decided to keep this as is